libs/arcade-mcp-server/arcade_mcp_server/server.py

```diff
@@ -754,8 +754,28 @@ async def _check_tool_requirements(
         tool_context: ToolContext,
         message: CallToolRequest,
         tool_name: str,
+        session: ServerSession | None = None,
     ) -> JSONRPCResponse[CallToolResult] | None:
         """Check tool requirements before executing the tool"""
+        # Check transport restrictions for tools requiring auth or secrets
+        if session and session.init_options:
+            transport_type = session.init_options.get("transport_type")
+            if transport_type != "stdio":
+                requirements = tool.definition.requirements
+                if requirements and (requirements.authorization or requirements.secrets):
+                    tool_response = {
+                        "message": (
+                            f"Tool '{tool_name}' cannot be executed over HTTP transport for security reasons. "
+                            "This tool requires end-user authorization or access to sensitive secrets, and therefore "
+                            "must not run over HTTP. This restriction is permanent - retrying the call will not resolve the issue.\n\n"
+                            "To use this tool, the server developer must either:\n"
+                            "1) Use the stdio transport (recommended for local development and desktop clients),\n"
+                            "2) Deploy the server using 'arcade deploy', or\n"
+                            "3) Register the server in the Arcade Developer Dashboard with type 'Arcade'."
```

Comment:
```suggestion
                            "3) Register the server in the Arcade Developer Dashboard with type 'Arcade' (rather than MCP), which also supports per-user OAuth."
```